### DISCUSSION TITLE
iOS에서 TextInput 수직중앙정렬이 안되는 이슈

### DISCUSSION THREAD

**Discussion by senasoon** [when-where/whenwhere-client#23] [Q&A]
@imironjin @sykang013 

iOS에서 TextInput 수직중앙정렬이 되지 않아 discussion 남깁니다.

![스크린샷 2023-06-28 오후 11 45 05](https://github.com/when-where/whenwhere-client/assets/102699437/84b23e1b-78f6-4f30-b5bb-98092445a5c2)

위 사진 왼쪽은 안드로이드, 오른쪽은 iOS입니다. 파란색 배경 부분이 TextInput 컴포넌트입니다.
안드로이드의 경우 텍스트 수직중앙정렬이 되지만 iOS의 경우 수직중앙정렬이 되지 않고 있습니다.

**Comment by imironjin** [comment]
혹시 css 코드 볼 수 있을까요 ?

**Reply by senasoon** [reply]
```
const StInput = styled.TextInput.attrs({
  placeholderTextColor: light.contents.contentThird,
})`
  padding: 0;
  background-color: blue;
  ${paragraph.Small}
`;
```

**Reply by imironjin** [reply]
관련 모든 css 코드를 봤으면 좋겠긴한데, input 안에 있는 저 두 개의 Text 를 Container로 감싸고,
 ```
 // Container
 display: flex;
 justify-content: center;
 align-items: center;
 ```
 이런식으로  Container css를 지정하고, 안에 Text를 넣어도 안될까요 ?

**Reply by senasoon** [reply]
넵 그렇습니다 🥲

**Comment by senasoon** [comment]
폰트 설정에 `line-height`가 있어 문제가 생긴 것을 확인했습니다.

관련된 이슈를 찾아보니 TextInput line-height 설정시 iOS에서 텍스트가 올바르게 정렬되지 않는 문제가 해결되지 않고 있는 것 같습니다.  
관련 이슈: https://github.com/facebook/react-native/issues/28012

`line-height: 0` 설정을 마지막에 주어 문제를 해결했습니다 🙂

**Comment by senasoon** [comment]
`line-height: 0` 을 사용하니 Android에서 텍스트가 보이지 않는 현상이 발생하여 react-native가 제공하는 Platform을 사용하여 Platform에 따라 다르게 스타일링을 주는 것으로 변경했습니다 🥲

```
// Platform import
import { Platform } from 'react-native';

// TextInput Component
 <StInput
  placeholder={placeholder}
  maxLength={maxLength}
  value={value}
  onChangeText={onChangeText}
/>
        

// TextInput Styling
const StInput = styled.TextInput.attrs({
  placeholderTextColor: light.contents.contentThird,
})`
  flex: 1;
  padding: 0;
  ${paragraph.Small};
  line-height: ${Platform.OS === 'ios' ? 0 : paragraph.Small.lineHeight};
`;
```
